### DISCUSSION TITLE
Alert user of virtualization of bootstrap script on Windows in powershell

### DISCUSSION THREAD

**Discussion by thatrobotguy** [microsoft/vcpkg#17907] [Ideas]
I have been messing with `vcpkg`s bootstrap lately trying to get `vcpkg.exe` to be created and I eventually came to my problem and solution, as outlined [here](https://github.com/microsoft/vcpkg/issues/9704). 

Essentially, my problem is that the bootstrap script kept getting flagged by anti-malware software and it would be blocked from actually writing any data to the hard drive.

The bootstrap script would correctly identify that the OS the script was running on was not recognized. This problem would make the script throw this error:
```
.\bootstrap-vcpkg.bat
Get-CimInstance : The specified service does not exist as an installed service.
At C:\Users\computeruser\vcpkg\scripts\bootstrap.ps1:349 char:20
+     $architecture=(Get-CimInstance win32_operatingsystem | Select-Obj ...
+                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (root\cimv2:win32_operatingsystem:String) [Get-CimInstance], CimException
    + FullyQualifiedErrorId : HRESULT 0x80070424,Microsoft.Management.Infrastructure.CimCmdlets.GetCimInstanceCommand

The property 'osarchitecture' cannot be found on this object. Verify that the property exists.
At C:\Users\computeruser\vcpkg\scripts\bootstrap.ps1:349 char:5
+     $architecture=(Get-CimInstance win32_operatingsystem | Select-Obj ...
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict
```
I am requesting a feature be built in to `vcpkg` such that it can recognize if the program is being containerized. I do not want other people after me to be stuck with correct scripts not working because of a silly malware guard. Maybe the script can output to the terminal some message and some options for getting the script to run outside of a container. A way to turn off that container through CLI would be great (though I do not know how possible that is; it may not be).

I am not sure how to implement this, but I wanted to make aware the developers of this software such that somebody at some point can make such a fix.

**Comment by vicroms** [comment]
Hi @thatrobotguy 

Thanks for the report! I'm glad you managed to pinpoint your issue. We're not currently looking into implementing virtualization detection but we would gladly take a PR from a contributor that implements the requested feature.
